ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4129,6 +4129,8 @@ public List<Partition> getPartitions(Table tbl) throws HiveException {
           GetPartitionsPsWithAuthResponse res = getMSC().listPartitionsWithAuthInfoRequest(req);
           tParts = res.getPartitions();
 
+        } catch (NoSuchObjectException nsoe) {
+          return Lists.newArrayList();
```

Comment:
For NoSuchObjectException, I think it would be ideal to throw the error message rather than returning an empty result.

Reply:
There are some cases in `Hive` throws empty result in case of `NoSuchObjectException`, for example:
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L3696
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L4058
The `TestHiveMetaStoreClientApiArgumentsChecker` tests the case

Reply:
The call `getMSC().listPartitionsWithAuthInfoRequest` here will goes to `ObjectStore. getPartitionsWithAuth`. The `getPartitionsWithAuth` doesn't check the table exits or not, it will return an empty list to the caller.
`
I merge the ObjectStore `getPartitionsWithAuth` and `listPartitionsPsWithAuth`, the two methods almost do the same thing except the `listPartitionsPsWithAuth` gets the partitions with partition values.

`listPartitionsPsWithAuth` does the check, and throws the exception in case the table is missing:
https://github.com/apache/hive/blob/84583f6fa1c779908852c49510c03495534f2e03/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L3916-L3918

`listPartitionsPsWithAuth` will use direct sql to get the partitions in this case, this will get some performance gain compared to `getPartitionsWithAuth`.